src/coreclr/src/vm/eventpipebuffermanager.cpp

```diff
@@ -168,6 +168,9 @@ EventPipeBuffer* EventPipeBufferManager::AllocateBufferForThread(EventPipeThread
         const unsigned int maxBufferSize = 1024 * 1024;
         bufferSize = Min(bufferSize, maxBufferSize);
 
+        // Make the buffer size fit into with pagesize-aligned block.
+        bufferSize = bufferSize + (g_SystemInfo.dwAllocationGranularity - (bufferSize % g_SystemInfo.dwAllocationGranularity));
```

Comment:
If buffersize is already a multiple of dwAllocationGranularity this rounds it up to the next multiple which is probably not what we want. I'd recommend copying the [GC's align up calculation](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/vm/gcenv.os.cpp#L341) for this:

````
size_t aligned_size = (size + g_SystemInfo.dwAllocationGranularity - 1) & ~static_cast<size_t>(g_SystemInfo.dwAllocationGranularity - 1);
````